src/adapter/src/flags.rs

```diff
@@ -44,19 +44,30 @@ pub fn compute_config(config: &SystemVars) -> ComputeParameters {
 }
 
 fn parse_yield_spec(s: &str) -> Option<YieldSpec> {
-    let parts: Vec<_> = s.split(':').collect();
-    match &parts[..] {
-        ["work", amount] => {
-            let amount = amount.parse().ok()?;
-            Some(YieldSpec::ByWork(amount))
-        }
-        ["time", millis] => {
-            let millis = millis.parse().ok()?;
-            let duration = Duration::from_millis(millis);
-            Some(YieldSpec::ByTime(duration))
+    let mut after_work = None;
+    let mut after_time = None;
+
+    let options = s.split(',').map(|o| o.trim());
+    for option in options {
+        let parts: Vec<_> = option.split(':').collect();
```

Comment:
Should we `trim` here, too? Someone might write `time: 1000`, note the space after the colon.

Reply:
Yep, can't hurt!